bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/WriteAdvHandle.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.bookkeeper.client.api;
+
+import io.netty.buffer.ByteBuf;
+import java.util.concurrent.CompletableFuture;
+import org.apache.bookkeeper.client.LedgerHandleAdv;
+import org.apache.bookkeeper.proto.DataFormats.LedgerMetadataFormat.DigestType;
+
+/**
+ * Provide write access to a ledger. Using WriteAdvHandler the writer MUST explictly set an entryId.
+ * Beware that the write for a given entryId will be acknowledged if and only if all entries up to entryId - 1 have
+ * been acknowledged too (expected from entryId 0)
+ *
+ * @see WriteHandler
+ */
+public interface WriteAdvHandle extends ReadHandle {
+
+    /**
+     * Add entry asynchronously to an open ledger
+     *
+     * @param entryId entryId to be added
+     * @param data array of bytes to be written to the ledger
+     * @return an handle to the result, in case of success it will return the same value of param entryId
+     */
+    public CompletableFuture<Long> write(final long entryId, byte[] data);
+
+    /**
+     * Add entry asynchronously to an open ledger
+     *
+     * @param entryId entryId to be added.
+     * @param data array of bytes to be written to the ledger
+     * @param offset offset from which to take bytes from data
+     * @param length number of bytes to take from data
+     * @return an handle to the result, in case of success it will return the same value of param entryId
+     */
+    public CompletableFuture<Long> write(final long entryId, byte[] data, int offset, int length);
```

Comment:
Can we drop the method of bytes? just keep the `ByteBuf` method. because if you have a byte[] array, you can always construct a `ByteBuf` using Unpooled.wrapBuffer(..).

Reply:
done, added ByteBuffer